## ТЕПЛОВА РОБОТА КОНВЕРТЕРНОГО АГРЕГАТУ І ШИХТОВКА ПЛАВКИ

## 1) *Теплоємність речовин*

Питома теплоємність – це кількість теплоти, яку необхідно надати 1 кг речовини для збільшення його температури на 1 °С (1 K). Це фундаментальна характеристика речовини, яка визначає її здатність поглинати та віддавати теплову енергію.

Одиниці вимірювання теплоємності:
- Питома теплоємність: [Дж/(кг·K)] - кількість теплоти, необхідна для нагрівання 1 кг речовини на 1 К
- Об'ємна теплоємність: [Дж/(м³·K)] - кількість теплоти, необхідна для нагрівання 1 м³ речовини на 1 К
- Мольна теплоємність: [Дж/(моль·K)] - кількість теплоти, необхідна для нагрівання 1 моль речовини на 1 К
  
Для переходу між цими трьома величина використовують густину речовини, її молярну масу та молярний об'єм. Наприклад, щоб перейти від мольної теплоємності до питомої, потрібно поділити мольну теплоємність на молярну масу речовини:
$$ c_{пит} = \frac{c_{мольн}}{M}, \text{ Дж/(кг·K)}, $$
де M - молярна маса речовини, кг/моль.

Для переходу від питомої до об'ємної теплоємності використовують густину речовини:
$$ c_{об} = c_{пит} \cdot \rho, \text{ Дж/(м³·K)}, $$
де ρ - густина речовини, кг/м³.

У теплових розрахунках використовують або істинні теплоємності $c_{icm}$, тобто теплоємності речовин при даній температурі, або середні теплоємності c в межах заданих температур $t_2-t_1$. У дослідах зазвичай знаходять середню величину теплоємності в даному інтервалі температур. Припустимо, що для нагрівання 1 моль речовини від $Т_1$ до $Т_2$ знадобилась теплота *Q*. Тоді середня теплоємність

$$ \bar{c} = \frac{Q}{T_2 - T_1}, \text{ Дж/(моль}\cdot\text{К)}. \qquad (1.1) $$

Фізичний зміст цієї формули: середня теплоємність є співвідношенням загальної кількості теплоти, необхідної для нагрівання речовини в заданому інтервалі температур, до різниці цих температур.

Істинна теплоємність є границею величини середньої теплоємності, коли різниця температур наближається до нескінченно малої величини *ΔТ*:

$$ c_{icm} = \lim_{\Delta T \to 0} \frac{Q}{\Delta T} = \frac{dQ}{dT}, \text{ Дж/(моль}\cdot\text{К)}. \qquad (1.2) $$

Тобто, істинна теплоємність - це теплоємність речовини при конкретній температурі. Вона представляє собою похідну від кількості теплоти по температурі. З точки зору фізики, істинна теплоємність показує, наскільки швидко змінюється теплота при зміні температури в конкретній точці.

При розрахунках теплових балансів металургійних процесів звичайно користуються табличними даними середніх теплоємностей речовин від нуля до певної температури *t*. Точне значення середньої теплоємності для будь-яких меж температури обчислюють, використовуючи математичну залежність істинної або іншої теплоємності від температури. Ця залежність виражається поліномом виду:



$$ c_{icm} = a_0 + a_1T + a_2T^2 + ... + a_nT^n, \text{ Дж/(моль}\cdot\text{К)}. \qquad (1.3) $$

де $a_o, a_1, a_2$ – коефіцієнти, які визначаються експериментальним шляхом.

Ці коефіцієнти отримують шляхом апроксимації експериментальних даних за методом найменших квадратів або іншими методами статистичної обробки. Кожен коефіцієнт має свій фізичний зміст:
- $a_0$ - константа, що відповідає теплоємності при температурі близькій до 0 К
- $a_1$ - коефіцієнт, що відображає лінійну зміну теплоємності з температурою
- $a_2$ - коефіцієнт, що відображає квадратичну зміну теплоємності з температурою

Для розрахунків зазвичай досить 2-3 членів цього поліному. У випадку підвищення температури від $Т_1$ до $Т_2$ кількість тепла, що поглинається визначається за формулою

$$ Q = \int_{T_1}^{T_2} c_{icm} dT, \text{ Дж/моль}. \qquad (1.4) $$

Пояснення: оскільки $c_{icm}$ - це кількість теплоти при нескінченно малій зміні температури ($dT$), то щоб знайти загальну кількість теплоти при нагріванні від $T_1$ до $T_2$, потрібно проінтегрувати $c_{icm}$ по всьому діапазону температур. Інтеграл $\int_{T_1}^{T_2} c_{icm} dT$ математично описує суму всіх нескінченно малих порцій теплоти, необхідних для нагрівання речовини від $T_1$ до $T_2$.

Підставляючи формулу (1.3) істинної теплоємності у формулу (1.4), отримуємо

$$ Q = \int_{T_1}^{T_2} (a_0 + a_1T + a_2T^2 + ... + a_nT^n) dT, \text{ Дж/моль}. \qquad (1.5) $$

Розглянемо детальніше процес інтегрування:

$$ Q = \int_{T_1}^{T_2} a_0 dT + \int_{T_1}^{T_2} a_1T dT + \int_{T_1}^{T_2} a_2T^2 dT + ... + \int_{T_1}^{T_2} a_nT^n dT $$

Виконаємо інтегрування кожного доданка:

$$ \int_{T_1}^{T_2} a_0 dT = a_0 \cdot (T_2 - T_1) $$
$$ \int_{T_1}^{T_2} a_1T dT = a_1 \cdot \frac{T_2^2 - T_1^2}{2} $$
$$ \int_{T_1}^{T_2} a_2T^2 dT = a_2 \cdot \frac{T_2^3 - T_1^3}{3} $$

І так далі для кожного члена поліному.

Такий самий результат можна одержати з виразу (1.1) для середньої теплоємності:

$$ Q = \bar{c}(T_2 - T_1), \text{ Дж/моль}. \qquad (1.6) $$

Ця формула є простішою, але вона потребує знання середньої теплоємності $\bar{c}$ в діапазоні температур від $T_1$ до $T_2$.

Прирівнявши праві частини рівнянь (1.5) та (1.6), отримаємо

$$ \bar{c}(T_2 - T_1) = \int_{T_1}^{T_2} (a_0 + a_1T + a_2T^2 + ... + a_nT^n) dT, \text{ Дж/моль}, \qquad (1.7) $$

або

$$ \bar{c}(T_2 - T_1) = a_0(T_2 - T_1) + a_1\frac{T_2^2 - T_1^2}{2} + a_2\frac{T_2^3 - T_1^3}{3} + ... + a_n\frac{T_2^{n+1} - T_1^{n+1}}{n+1}, \text{ Дж/моль}, \qquad (1.8) $$

Це рівняння можна спростити, розділивши обидві частини на $(T_2 - T_1)$:

$$ \bar{c} = a_0 + a_1\frac{T_2^2 - T_1^2}{2(T_2 - T_1)} + a_2\frac{T_2^3 - T_1^3}{3(T_2 - T_1)} + ... + a_n\frac{T_2^{n+1} - T_1^{n+1}}{(n+1)(T_2 - T_1)}, $$

Далі, спростимо алгебраїчні вирази:

$$ \frac{T_2^2 - T_1^2}{2(T_2 - T_1)} = \frac{(T_2 - T_1)(T_2 + T_1)}{2(T_2 - T_1)} = \frac{T_2 + T_1}{2} $$

$$ \frac{T_2^3 - T_1^3}{3(T_2 - T_1)} = \frac{(T_2 - T_1)(T_2^2 + T_1T_2 + T_1^2)}{3(T_2 - T_1)} = \frac{T_2^2 + T_1T_2 + T_1^2}{3} $$

І так далі для кожного члена. Підставляючи ці вирази, отримуємо:

$$ \bar{c} = a_0 + a_1\frac{T_2 + T_1}{2} + a_2\frac{T_2^2 + T_1T_2 + T_1^2}{3} + ... + a_n\frac{T_2^n + T_2^{n-1}T_1 + ... + T_1^n}{n+1}, \text{ Дж/(моль}\cdot\text{К)}. \qquad (1.9) $$



Цей вираз дозволяє розрахувати середню теплоємність речовини в діапазоні температур від $T_1$ до $T_2$, знаючи коефіцієнти полінома, що описує залежність істинної теплоємності від температури.

Для деяких речовин у довідковій літературі наводять температурну залежність середніх теплоємностей із вказівкою інтервалів температур, тоді відпадає необхідність в інтегруванні з урахуванням залежності істинної теплоємності речовин від температури. Для визначення істинної теплоємності з середньої у межах від 0 до *Т* необхідно помножити середню теплоємність на температуру і виконати диференціювання по *Т*:

$$ c_{icm} = \frac{d(\bar{c} \cdot T)}{dT}, \text{ Дж/(моль}\cdot\text{К)}. \qquad (1.10) $$

Розглянемо доведення цієї формули. За визначенням, середня теплоємність від 0 до T:

$$ \bar{c} = \frac{Q}{T - 0} = \frac{Q}{T} $$

Звідси:
$$ Q = \bar{c} \cdot T $$

З іншого боку, за визначенням істинної теплоємності:
$$ c_{icm} = \frac{dQ}{dT} $$

Підставляючи вираз для Q, отримуємо:
$$ c_{icm} = \frac{d(\bar{c} \cdot T)}{dT} $$

Що і потрібно було довести.

За відсутності таких даних теплоємність твердих речовин обчислюють по емпіричній формулі, відповідно до якої теплоємність твердих речовин підкоряється правилу адитивності, тобто дорівнює алгебраїчній сумі теплоємностей компонентів, що входять у дану речовину (суміш):

$$ c_{сум} = \sum n_ic_i, \text{ Дж/(кг}\cdot\text{К)}, \qquad (1.11) $$

де 
- $n_і$ – частка *і*-ого компонента, що входить у дану речовину; *(в залежності від того, яку теплоємність необхідно розрахувати (масову, об’ємну чи мольну), використовується відповідна частка.)*
- $с_і$ – теплоємність *і*-ого компонента суміші, Дж/(кг·K).


Правило адитивності базується на експериментальних спостереженнях і застосовується як до твердих, так і до рідких та газоподібних сумішей. Наприклад, для обчислення масової теплоємності суміші:

$$ c_{сум} = \sum_{i=1}^{n} w_i c_i $$

де $w_i$ - масова частка і-го компонента (в долях одиниці), $c_i$ - питома теплоємність і-го компонента.

Для мольної теплоємності суміші:

$$ c_{сум} = \sum_{i=1}^{n} x_i c_i $$

де $x_i$ - мольна частка і-го компонента (в долях одиниці), $c_i$ - мольна теплоємність і-го компонента.

У технологічних розрахунках широко застосовують метод визначення кількості поглиненої або виділеної в тому чи іншому процесі теплоти за тепловмістом даної речовини. При нагріванні одиниці маси речовини (кг, моль тощо) від $Т_1$ до $Т_2$ за сталого тиску (*Р = const*)


$$ Q_p = \bar{c}_p(T_2 - T_1) = \bar{c}_pT_2 - \bar{c}_pT_1, \text{ Дж/кг}. \qquad (1.12) $$

Такий результат можна одержати, використовуючи для визначення кількості теплоти перший закон термодинаміки для ізобарного процесу:

$$Q_p = (U_к - U_п) + P(V_к - V_п) = (U_к + PV_к) - (U_п + PV_п),\text{ Дж/кг}. \qquad (1.13)$$

Тут необхідно пояснити фізичний зміст величин:
- $U_к$ і $U_п$ - внутрішня енергія речовини в кінцевому та початковому станах відповідно, Дж/кг
- $P$ - тиск, що залишається постійним протягом процесу, Па
- $V_к$ і $V_п$ - об'єм речовини в кінцевому та початковому станах відповідно, м³/кг

Перший закон термодинаміки встановлює, що теплота, підведена до системи, витрачається на зміну внутрішньої енергії системи та на виконання роботи системою проти зовнішніх сил. У випадку ізобарного процесу (P = const), робота дорівнює $P(V_к - V_п)$.



Позначимо $U_к + PV_к = H_к$ і $U_п + PV_п = H_п$ тоді

$$ Q_p = H_к - H_п, \text{ Дж/кг}. \qquad (1.14) $$

Функція $H = U + PV$ , що залежить тільки від стану системи, називається *ентальпією* або *тепловмістом*. Таким чином, якщо відомо ентальпію речовини за даних умов, то відпадає необхідність обчислювати теплоємність для тих же умов.

Ентальпія є важливою термодинамічною функцією стану, яка характеризує енергетичний вміст системи. Вона складається з внутрішньої енергії системи U (яка включає кінетичну енергію молекул, енергію їхньої взаємодії тощо) та добутку тиску P на об'єм V, який характеризує потенційну енергію системи через її здатність виконувати роботу за рахунок розширення.

Зміна ентальпії в ізобарному процесі (при постійному тиску) дорівнює кількості теплоти, переданої системі. Це робить ентальпію особливо корисною для розрахунків теплових ефектів хімічних реакцій та фазових переходів, які часто відбуваються при постійному тиску.

Величину ентальпія беруть з довідкової літератури. У розрахунках звичайно оперують не абсолютними, а відносними величинами ентальпії. Для цього умовно обирають нульовий стан системи, від якого й обчислюють її ентальпію. У нульовому стані ентальпія умовно приймається рівною нулю. Величину *Н* за температури *t, °С* беруть відповідно до теплових закономірностей даного тіла відносно цього нульового стану.

Найчастіше за нульовий стан приймають:
- для речовин при стандартних умовах (25°C і 1 атм) - стандартну ентальпію утворення
- для елементів у їх стандартних станах за тих же умов - нульову ентальпію

Зв'язок між ентальпією та теплоємністю при постійному тиску можна виразити формулою:

$$ c_p = \left(\frac{\partial H}{\partial T}\right)_p $$

Тобто, теплоємність при постійному тиску є частинною похідною ентальпії по температурі при постійному тиску.

Для визначення ентальпії газових сумішей користуються правилом змішування (адитивності):

$$ H_{сум} = 0,01 \sum_{i=1}^{k} n_i H_i = 0,01(n_1H_1 + n_2H_2 + ... + n_kH_k), \text{ Дж/кг}, \qquad (1.15) $$

де 
- $n_i$ – об’ємна частка *і*-ого газу в суміші, %;
- $H_i$ – ентальпія *і*-ого газу, Дж/кг. 

Коефіцієнт 0,01 використовується для переведення відсотків у частки одиниці. Це правило базується на принципі адитивності термодинамічних властивостей для ідеальних газів та сумішей, що підкоряються закону Дальтона. Згідно з цим принципом, кожний компонент суміші поводиться незалежно від інших компонентів, і загальна ентальпія суміші є сумою ентальпій окремих компонентів, зважених за їх часткою в суміші.

## Додаткова інформація про теплоємність

### Теплоємність при постійному об'ємі та постійному тиску

Існує два основних типи теплоємності:
1. Теплоємність при постійному об'ємі ($c_v$) - кількість теплоти, необхідна для підвищення температури речовини на 1 K при постійному об'ємі.
2. Теплоємність при постійному тиску ($c_p$) - кількість теплоти, необхідна для підвищення температури речовини на 1 K при постійному тиску.

Для ідеального газу зв'язок між цими теплоємностями визначається рівнянням Майєра:

$$ c_p - c_v = R $$

де R - універсальна газова стала (для мольної теплоємності) або індивідуальна газова стала (для питомої теплоємності).

Для реальних газів, рідин і твердих тіл співвідношення більш складне і залежить від їх термодинамічних властивостей.

### Закон Дюлонга-Пті

Для простих твердих тіл (особливо металів) мольна теплоємність при кімнатній температурі приблизно дорівнює 3R ≈ 25 Дж/(моль·K). Це емпіричне правило відоме як закон Дюлонга-Пті.

### Теплоємність при фазових переходах

При фазових переходах (плавлення, випаровування тощо) теплоємність речовини прямує до нескінченності, оскільки теплота витрачається на зміну фази без зміни температури. У термодинамічних розрахунках цю особливість враховують окремо через теплоту фазового переходу.

### Вплив температури на теплоємність

Для більшості речовин теплоємність залежить від температури. При низьких температурах (близьких до абсолютного нуля) теплоємність твердих тіл прямує до нуля згідно з третім законом термодинаміки. При підвищенні температури теплоємність зазвичай зростає і може бути описана різними емпіричними формулами, зокрема законом Дебая або законом Ейнштейна для кристалічних речовин.

## Термодинамічні процеси та їх теплові ефекти

### Ізохорний процес (V = const)

В ізохорному процесі об'єм системи залишається постійним, і робота розширення дорівнює нулю. Тому перший закон термодинаміки приймає вигляд:

$$ Q_v = U_к - U_п = \Delta U $$

Тобто, теплота, передана системі, повністю йде на зміну її внутрішньої енергії.

### Ізобарний процес (P = const)

В ізобарному процесі тиск залишається постійним. Як було показано вище, теплота в цьому процесі дорівнює зміні ентальпії:

$$ Q_p = H_к - H_п = \Delta H $$

### Ізотермічний процес (T = const)

В ізотермічному процесі температура залишається постійною. Для ідеального газу зміна внутрішньої енергії дорівнює нулю, і теплота дорівнює роботі:

$$ Q_T = W = nRT\ln\frac{V_2}{V_1} = nRT\ln\frac{P_1}{P_2} $$

де n - кількість моль газу, R - універсальна газова стала.

### Адіабатний процес (Q = 0)

В адіабатному процесі теплообмін з навколишнім середовищем відсутній. Перший закон термодинаміки приймає вигляд:

$$ 0 = \Delta U + W $$

Тобто, робота виконується виключно за рахунок зміни внутрішньої енергії системи.

Для ідеального газу адіабатний процес описується рівнянням:

$$ PV^\gamma = const $$

де γ = c_p/c_v - показник адіабати.

### Задача №1: Визначення теплоти, що поглинає діоксид вуглецю при нагріванні

#### *Умова задачі*
За коефіцієнтами з таблиці 1.1 визначити теплоту, що поглинає діоксид вуглецю при нагріванні 100 м³ від 25 до 1000 °С за сталого об'єму.

#### Фізичний зміст задачі
Ця задача має важливе практичне значення в металургійних процесах, особливо в конвертерних агрегатах, де відбувається інтенсивне нагрівання газів. Діоксид вуглецю (CO₂) є одним із основних газів, що утворюються під час окислення вуглецю в сталеплавильних процесах. Розуміння теплових властивостей CO₂ дозволяє точніше розраховувати теплові баланси конвертерних процесів та оптимізувати їх енергоефективність.

#### Таблиця 1.1

Теплоємності деяких газів $c_p = a + bT + cT^{-2}$

| Газ | a | b·10³ | c·10⁻⁵ | ΔT, К |
|-----|-----|--------|---------|--------|
| CO | 6,79 | 0,98 | -0,11 | 298-2500 |
| CO₂ | 10,55 | 2,16 | -2,04 | 298-2500 |
| CH₄ | 5,65 | 11,44 | -0,46 | 298-1500 |
| C₂H₄ | - | - | - | - |
| H₂ | 27,30 | 3,27 | 0,50 | 298-3000 |
| H₂O | 18,03 | - | - | 273-373 |
| N₂ | 6,66 | 1,02 | - | 298-2500 |
| O₂ | 7,16 | 1,0 | -0,40 | 298-3000 |

*Примітка: Коефіцієнти наведені в кал/(моль·К)*

##### РОЗВ'ЯЗОК

#### Теплоємність та її види
Питома теплоємність – це кількість теплоти, яку необхідно надати 1 кг речовини для збільшення його температури на 1°С (1 K). Це одна з найважливіших теплофізичних характеристик речовини, яка визначає її здатність акумулювати теплову енергію.

Залежно від умов нагрівання розрізняють:
- **Теплоємність при сталому тиску ($c_p$)** — кількість теплоти, необхідна для нагрівання 1 моля речовини на 1 К при постійному тиску. При такому нагріванні газ розширюється і виконує роботу проти зовнішніх сил.
- **Теплоємність при сталому об'ємі ($c_v$)** — кількість теплоти, необхідна для нагрівання 1 моля речовини на 1 К при постійному об'ємі. При такому нагріванні газ не виконує роботи розширення.

Для ідеальних газів різниця між теплоємністю при сталому тиску ($c_p$) і теплоємністю при сталому об'ємі ($c_v$) дорівнює універсальній газовій сталій:
$$ c_p - c_v = R $$
де *R* - універсальна газова стала (R = 8,31446261815 Дж/(моль·К))

Це співвідношення випливає з першого закону термодинаміки та рівняння стану ідеального газу. Фізичний зміст цієї різниці полягає в тому, що при нагріванні за сталого тиску частина теплоти витрачається на виконання роботи розширення газу, тоді як при нагріванні за сталого об'єму вся теплота йде на збільшення внутрішньої енергії газу.



#### Залежність теплоємності від температури
Для газів залежність істинної теплоємності від температури виражається поліномом:
$$ c_p = a + bT + cT^{-2} $$

Кожен член цього поліному має своє фізичне обґрунтування:
- Константа **a** відображає внесок поступального та обертального руху молекул газу в теплоємність
- Член **bT** враховує вплив коливальних ступенів свободи молекул, які активуються при підвищенні температури
- Член **cT^{-2}** враховує квантові ефекти при низьких температурах та міжмолекулярну взаємодію

Ця емпірична формула дозволяє з високою точністю описувати теплоємність газів у широкому діапазоні температур, що особливо важливо для інженерних розрахунків у металургії.

#### Розрахунок кількості теплоти
Кількість теплоти при нагріванні за сталого об'єму визначається за формулою:
$$ Q = n\int_{T_1}^{T_2} c_v dT $$
де:
- *n* - кількість молів речовини
- $T_1$, $T_2$ - початкова і кінцева температура в Кельвінах
- $c_v$ - теплоємність при сталому об'ємі

Інтеграл у цій формулі представляє собою площу під кривою залежності теплоємності від температури в діапазоні від $T_1$ до $T_2$. Фізично це означає сумування нескінченно малих кількостей теплоти $dQ = nc_v dT$, необхідних для нагрівання речовини на нескінченно малі інтервали температури $dT$ у всьому діапазоні нагрівання.

#### Крок 1: Перетворення температур в Кельвіни
Оскільки всі термодинамічні формули використовують абсолютну шкалу температур (Кельвіни), необхідно перевести температури із шкали Цельсія в Кельвіни:
$$ T_1 = 25°C + 273,15 = 298,15\text{ К} $$
$$ T_2 = 1000°C + 273,15 = 1273,15\text{ К} $$

Важливо перевірити, чи входять ці температури в діапазон застосовності формули з таблиці 1.1. Для CO₂ діапазон становить 298-2500 К, тому обидві температури знаходяться в допустимому діапазоні, і ми можемо використовувати наведені коефіцієнти.

#### Крок 2: Розрахунок кількості молів газу

Для розрахунку кількості молів газу використовується рівняння стану ідеального газу:

$$ PV = nRT $$

звідки:

$$ n = \frac{PV}{RT} $$

де:
- *P* - тиск, Па
- *V* - об'єм, м³
- *R* - універсальна газова стала, Дж/(моль·К)
- *T* - температура, К

Рівняння стану ідеального газу описує зв'язок між макроскопічними параметрами газу (тиском, об'ємом, температурою) та кількістю речовини. Воно базується на кінетичній теорії газів і є фундаментальним законом термодинаміки.

Приймаємо стандартні умови для початкового стану газу:
- *P* = 101325 Па (нормальний атмосферний тиск)
- *V* = 100 м³ (задано в умові)
- *R* = 8,31446261815 Дж/(моль·К) (універсальна газова стала)
- *T* = $T_1$ = 298,15 К (початкова температура)

Звідси розраховуємо кількість речовини:
$$ n = \frac{PV}{RT_1} = \frac{101325 \cdot 100}{8,31446261815 \cdot 298,15} = 4087,40\text{ моль} $$

Ця кількість молів залишається незмінною протягом усього процесу нагрівання, оскільки маса газу не змінюється (закон збереження маси).

#### Крок 3: Перетворення коефіцієнтів $c_p$ в $c_v$
Оскільки в таблиці 1.1 наведені коефіцієнти для теплоємності при сталому тиску ($c_p$), а нам потрібна теплоємність при сталому об'ємі ($c_v$), використовуємо співвідношення $c_p - c_v = R$:

1. Переведемо R в кал/(моль·К) для узгодження одиниць вимірювання з коефіцієнтами з таблиці:
   $$ R = 8,31446261815 \text{ Дж/(моль·К)} = 1,986 \text{ кал/(моль·К)} $$
   Коефіцієнт перерахунку: 1 кал = 4,1868 Дж

2. Розрахуємо коефіцієнти для cv:
   $$ c_v = c_p - R $$
   
   Для коефіцієнта при T⁰:
   $$ a_{(v)} = a - R = 10,55 - 1,987 = 8,564 \text{ кал/(моль·К)} $$
   
   Для коефіцієнта при T¹:
   $$ b_{(v)} = b = 2,16 / 10^{3} = 0,00216 \text{ кал/(моль·К²)} $$
   
   Для коефіцієнта при T⁻²:
   $$ c_{(v)} = c = -2,04 / 10^{-5} = -204000 \text{ кал·К/(моль)} $$

Зверніть увагу, що різниця $c_p - c_v = R$ впливає лише на коефіцієнт a, оскільки R є константою і не залежить від температури. Коефіцієнти при T¹ та T⁻² залишаються незмінними.


#### Крок 4: Переведення коефіцієнтів в Дж/(моль·К)
Для подальших розрахунків переводимо коефіцієнти з кал/(моль·К) в Дж/(моль·К), множачи кожен коефіцієнт на 4,1868 Дж/кал:

$$ a_{(v)} = 8,564 \cdot 4,1868 = 35,86 \text{ Дж/(моль·К)} $$
$$ b_{(v)} = 0,00216 \cdot 4,1868 = 0,009044 \text{ Дж/(моль·К²)} $$
$$ c_{(v)} = -204000 \cdot 4,1868 = -854107,2 \text{ Дж·К/(моль)} $$

Тепер теплоємність при сталому об'ємі виражається формулою:
$$ c_v = 35,85 + 0,009044 \cdot T - 854107,2 \cdot T^{-2} \text{ Дж/(моль·К)} $$

#### Крок 5: Розрахунок теплоти
Для розрахунку кількості теплоти необхідно проінтегрувати вираз для теплоємності в діапазоні температур:

$$ Q = n\int_{T_1}^{T_2} c_v dT = n\int_{T_1}^{T_2} (a_{(v)} + b_{(v)}T + c_{(v)}T^{-2}) dT $$

Розбиваємо інтеграл на три частини для зручності обчислення:

1. Інтегрування константи $a_v$:
   $$ \int_{T_1}^{T_2} a_{(v)} dT = a_{(v)}(T_2 - T_1) $$

2. Інтегрування лінійного члена $b_vT$:
   $$ \int_{T_1}^{T_2} b_{(v)}T dT = b_{(v)}\frac{T^2}{2}\bigg|_{T_1}^{T_2} = b_{(v)}\frac{T_2^2 - T_1^2}{2} $$

3. Інтегрування оберненого квадрата $c_vT^{-2}$:
   $$ \int_{T_1}^{T_2} c_{(v)}T^{-2} dT = -c_{(v)}\frac{1}{T}\bigg|_{T_1}^{T_2} = c_{(v)}(-\frac{1}{T_2} + \frac{1}{T_1}) $$

Об'єднуючи всі частини, отримуємо:

$$ Q = n[a_{(v)}(T_2 - T_1) + \frac{b_{(v)}}{2}(T_2^2 - T_1^2) + c_{(v)}(-\frac{1}{T_2} + \frac{1}{T_1})] $$

Підставляємо числові значення:

$$ Q = 4087,40 \cdot [35,86(1273,15 - 298,15) + \frac{9,044 \cdot 10^{-3}}{2}(1273,15^2 - 298,15^2) + (-8,541 \cdot 10^{5})(-\frac{1}{1273,15} + \frac{1}{298,15})] $$

$$ Q = 4087,40 \cdot [35,86 \cdot 975 + \frac{0,009044}{2}(1620910,92 - 88893,42) + (-854107,2)(-0,000785 + 0,003354)] $$

$$ Q = 4087,40 \cdot [34953,75 + 0,00452 \cdot 1532017,5 + (-854107,2 \cdot 0,002569] $$

$$ Q = 4087,40 \cdot [34959,87 + 6927,39 - 2193.83] $$

$$ Q = 4087,40 \cdot 39693,43 = 162243115,62 \text{ Дж} = 162243,12 \text{ кДж} $$

### Відповідь
Теплота, що поглинає діоксид вуглецю при нагріванні 100 м³ від 25 до 1000 °С за сталого об'єму, становить Q ≈ 162243,12 кДж.

### Фізична інтерпретація результату
Отримане значення теплоти (162243,12 кДж) показує, скільки теплової енергії необхідно надати 100 м³ діоксиду вуглецю, щоб нагріти його від 25°C до 1000°C при сталому об'ємі. Ця енергія витрачається на збільшення внутрішньої енергії газу, що проявляється у збільшенні кінетичної енергії молекул CO₂ (зростає швидкість їх поступального, обертального та коливального рухів).

Для порівняння, ця кількість теплоти еквівалентна енергії, що виділяється при спалюванні приблизно 3 кг природного газу.

In [1]:
"""
Задача №1: Визначення теплоти, що поглинає діоксид вуглецю при нагріванні

Умова: За коефіцієнтами з таблиці 1.1 визначити теплоту, що поглинає діоксид
вуглецю при нагріванні 100 м³ від 25 до 1000 °С за сталого об'єму.

Теоретичні відомості:
1. Істинна теплоємність (формула 1.2):
   c_icm = lim(ΔT→0) ΔQ/ΔT = dQ/dT
   
   Пояснення: Істинна теплоємність - це границя відношення кількості теплоти до зміни 
   температури при нескінченно малій зміні температури. Фактично це похідна від кількості 
   теплоти за температурою, яка показує, як швидко змінюється теплота при зміні температури 
   в конкретній точці.

2. Залежність теплоємності від температури (формула 1.3):
   c_icm = a₀ + a₁T + a₂T² + ... + aₙTⁿ
   
   Пояснення: Теплоємність не є константою, а залежить від температури. Ця залежність 
   описується поліномом, коефіцієнти якого визначаються експериментально. Для газів 
   часто використовують спрощену формулу c_p = a + bT + cT^(-2).

3. Кількість теплоти (формула 1.4):
   Q = ∫(T1 до T2) c_icm dT
   
   Пояснення: Щоб знайти загальну кількість теплоти при нагріванні від T1 до T2, 
   потрібно проінтегрувати теплоємність по всьому діапазону температур. Інтеграл 
   представляє собою суму всіх нескінченно малих порцій теплоти.

4. Зв'язок між cp і cv:
   cp - cv = R, де R - універсальна газова стала
   
   Пояснення: Для ідеальних газів різниця між теплоємністю при сталому тиску (cp) і 
   теплоємністю при сталому об'ємі (cv) дорівнює універсальній газовій сталій. Це 
   пов'язано з тим, що при нагріванні за сталого тиску газ виконує додаткову роботу 
   розширення, на яку витрачається частина теплоти.

Довідкові дані:
1. Коефіцієнти для CO₂ з таблиці 1.1:
   a = 10.55 кал/(моль·К) - коефіцієнт при T⁰, відображає внесок поступального та обертального руху молекул
   b = 2.16/10³ кал/(моль·К²) - коефіцієнт при T¹, враховує вплив коливальних ступенів свободи
   c = -2.04/10⁻⁵ кал·К/(моль) - коефіцієнт при T⁻², враховує квантові ефекти та міжмолекулярну взаємодію
   Діапазон температур: 298-2500 К - межі застосовності формули

2. Фізичні константи:
   R = 8.31446261815 Дж/(моль·К) - універсальна газова стала
   1 кал = 4.1868 Дж - коефіцієнт перетворення калорій в джоулі
"""

# Імпортуємо бібліотеку NumPy для роботи з масивами та математичними функціями
# NumPy (Numerical Python) - це бібліотека для наукових обчислень, яка містить:
# - Багатовимірні масиви (ndarray)
# - Математичні функції для роботи з масивами
# - Функції для числового інтегрування, диференціювання, тощо
import numpy as np

In [2]:
# Фізичні константи
# Визначаємо основні фізичні константи, які будуть використовуватися в розрахунках

# Універсальна газова стала - фундаментальна фізична константа, яка зв'язує тиск,
# об'єм, кількість речовини та температуру ідеального газу через рівняння стану PV = nRT
R = 8.31446261815  # Дж/(моль·К)

# Коефіцієнт перетворення калорій в джоулі
# Калорія - одиниця вимірювання енергії
# 1 калорія = 4.1868 джоулів (точне значення за міжнародним стандартом)
CAL_TO_JOULE = 4.1868  # Дж/кал

# Стандартний атмосферний тиск - тиск сухого повітря на рівні моря при 0°C
# Використовується як опорне значення в багатьох термодинамічних розрахунках
P_STANDARD = 101325  # Па (1 атм)

# Температура за шкалою Кельвіна, що відповідає 0°C
# Абсолютний нуль за шкалою Кельвіна становить -273.15°C
# Ця константа дозволяє переводити температуру з градусів Цельсія в Кельвіни
T_STANDARD = 273.15  # К (0°C)

In [3]:
# Коефіцієнти для CO₂ з таблиці 1.1
# Ці коефіцієнти описують залежність теплоємності CO₂ від температури
# за формулою cp = a + bT + cT^(-2)

# Коефіцієнт a - вільний член у формулі теплоємності
# Фізично відображає внесок поступального та обертального руху молекул газу
a = 10.55  # кал/(моль·К)

# Коефіцієнт b - коефіцієнт при T у формулі теплоємності
# Відображає вплив коливальних ступенів свободи молекул, які активуються при підвищенні температури
# Запис 2.16e-3 означає 2.16·10⁻³ (наукова нотація в Python)
b = 2.16e-3  # кал/(моль·К²)

# Коефіцієнт c - коефіцієнт при T^(-2) у формулі теплоємності
# Враховує квантові ефекти при низьких температурах та міжмолекулярну взаємодію
# Від'ємне значення вказує на зменшення теплоємності при дуже низьких температурах
c = -2.04e5  # кал·К²/(моль)

In [4]:
# Діапазон температур для коефіцієнтів
# Визначаємо межі застосовності формули теплоємності з таблиці 1.1
# Використання коефіцієнтів поза цим діапазоном може призвести до неточних результатів

# Мінімальна температура діапазону (25°C в Кельвінах)
T_MIN = 298  # К

# Максимальна температура діапазону
# Верхня межа застосовності формули для CO₂
T_MAX = 2500  # К

In [5]:
# Вхідні дані
# Параметри, задані в умові задачі

# Об'єм газу, який потрібно нагріти
# Це один з ключових параметрів, що визначає кількість речовини
volume = 100  # м³

# Початкова температура газу за шкалою Цельсія
# Температура, від якої починається нагрівання
t1 = 25  # °C

# Кінцева температура газу за шкалою Цельсія
# Температура, до якої потрібно нагріти газ
t2 = 1000  # °C

In [6]:
# 1. Перетворення температур в Кельвіни
# Всі термодинамічні формули використовують абсолютну шкалу температур (Кельвіни)
# Для переведення з градусів Цельсія в Кельвіни додаємо 273.15

# Перетворення початкової температури
T1 = t1 + T_STANDARD  # T1 = 25 + 273.15 = 298.15 К

# Перетворення кінцевої температури
T2 = t2 + T_STANDARD  # T2 = 1000 + 273.15 = 1273.15 К

# Виведення результатів перетворення для перевірки
# Функція print() виводить інформацію на екран
# f-рядки (f"...") дозволяють вставляти змінні безпосередньо в текст
# {T1:.2f} означає, що T1 буде виведено з двома знаками після коми
print(f"1. Температури:")
print(f"   T1 = {t1}°C = {T1:.2f} К")
print(f"   T2 = {t2}°C = {T2:.2f} К")

1. Температури:
   T1 = 25°C = 298.15 К
   T2 = 1000°C = 1273.15 К


In [7]:
# 2. Перевірка діапазону температур
# Важливо перевірити, чи входять температури в діапазон застосовності формули
# Якщо температури виходять за межі діапазону, результати можуть бути неточними

# Цикл for перебирає обидві температури (T1 і T2)
for T in [T1, T2]:
    # Перевіряємо, чи температура знаходиться в допустимому діапазоні
    # not (умова) - заперечення умови
    # T_MIN <= T <= T_MAX - перевірка, чи T знаходиться між T_MIN і T_MAX
    if not (T_MIN <= T <= T_MAX):
        # Якщо температура виходить за межі діапазону, виводимо попередження
        print(f"\nУВАГА: Температура {T:.2f} К виходить за межі діапазону {T_MIN}-{T_MAX} К")
        print("Результати можуть бути неточними!")

In [8]:
# 3. Розрахунок кількості молів за рівнянням стану ідеального газу
# Для розрахунку кількості теплоти потрібно знати кількість речовини (молів)
# Використовуємо рівняння стану ідеального газу: PV = nRT

# Виражаємо n з рівняння: n = PV/(RT)
# P - тиск (P_STANDARD)
# V - об'єм (volume)
# R - універсальна газова стала (R)
# T - температура (T1) - використовуємо початкову температуру, оскільки
#     кількість молів визначається для початкового стану газу
n = (P_STANDARD * volume) / (R * T1)

# Виведення результату розрахунку
print(f"\n2. Кількість речовини:")
print(f"   n = {n:.2f} моль")


2. Кількість речовини:
   n = 4087.40 моль


In [9]:
# 4. Розрахунок коефіцієнтів для cv (теплоємності при сталому об'ємі)
# В таблиці 1.1 наведені коефіцієнти для cp (теплоємності при сталому тиску)
# Для розрахунку теплоти при сталому об'ємі потрібно перейти до cv

# Конвертуємо універсальну газову сталу R з Дж/(моль·К) в кал/(моль·К)
# Це потрібно для узгодження одиниць вимірювання з коефіцієнтами з таблиці
R_cal = R / CAL_TO_JOULE  # R_cal = 8.31446261815 / 4.1868 ≈ 1.987 кал/(моль·К)

In [10]:
# Розрахунок коефіцієнтів для cv за формулою cv = cp - R
# Різниця між cp і cv дорівнює R тільки для коефіцієнта при T⁰
# Коефіцієнти при інших степенях T залишаються незмінними

# Коефіцієнт при T⁰ для cv
a_v = a - R_cal  # a_v = 10.55 - 1.987 = 8.563 кал/(моль·К)

# Коефіцієнт при T¹ для cv (залишається таким самим, як для cp)
b_v = b  # b_v = 0.00216 кал/(моль·К²)

# Коефіцієнт при T⁻² для cv (залишається таким самим, як для cp)
c_v = c  # c_v = -2.04·10⁵ кал·К²/(моль)

In [11]:
# Переводимо коефіцієнти з кал/(моль·К) в Дж/(моль·К)
# Це потрібно для подальших розрахунків, щоб отримати результат в джоулях
# Множимо кожен коефіцієнт на коефіцієнт перетворення CAL_TO_JOULE (4.1868 Дж/кал)

# Коефіцієнт при T⁰ для cv в Дж/(моль·К)
a_v *= CAL_TO_JOULE  # a_v = 8.563 * 4.1868 = 35.85 Дж/(моль·К)

# Коефіцієнт при T¹ для cv в Дж/(моль·К²)
b_v *= CAL_TO_JOULE  # b_v = 0.00216 * 4.1868 = 9.044·10⁻³ Дж/(моль·К²)

# Коефіцієнт при T⁻² для cv в Дж·К²/(моль)
c_v *= CAL_TO_JOULE  # c_v = -2.04·10⁵ * 4.1868 = -8.541·10⁵ Дж·К²/(моль)

### Аналітичний розрахунок теплоти
При розрахунку кількості теплоти, що поглинається газом при нагріванні, будемо використовувати формулу:

$$ Q = n\int_{T_1}^{T_2} c_v \, dT $$

де:
- *Q* - кількість теплоти, Дж
- *n* - кількість речовини, моль
- *T₁*, *T₂* - початкова та кінцева температура, К
- *cᵥ* - теплоємність при сталому об'ємі, Дж/(моль·К)

Для газів теплоємність залежить від температури за формулою:

$$ c_v = a_{(v)} + b_{(v)}T + c_{(v)}T^{-2} $$

де:
- *aᵥ*, *bᵥ*, *cᵥ* - коефіцієнти з таблиці 1.1
- *T* - температура, К

#### Інтегрування

Підставляємо вираз для теплоємності в формулу для теплоти:

$$ Q = n\int_{T_1}^{T_2} (a_{(v)} + b_{(v)}T + c_{(v)}T^{-2}) \, dT $$

Розбиваємо інтеграл на три частини:

$$ Q = n[\int_{T_1}^{T_2} a_{(v)} \, dT + \int_{T_1}^{T_2} b_{(v)}T \, dT + \int_{T_1}^{T_2} c_{(v)}T^{-2} \, dT] $$

##### 1. Інтегрування константи

$$ \int_{T_1}^{T_2} a_{(v)} \, dT = a_{(v)}(T_2 - T_1) $$

##### 2. Інтегрування лінійного члена

$$ \int_{T_1}^{T_2} b_{(v)}T \, dT = b_{(v)}\frac{T^2}{2}\bigg|_{T_1}^{T_2} = b_{(v)}\frac{T_2^2 - T_1^2}{2} $$

##### 3. Інтегрування оберненого квадрата

$$ \int_{T_1}^{T_2} c_{(v)}T^{-2} \, dT = -c_{(v)}\frac{1}{T}\bigg|_{T_1}^{T_2} = c_{(v)}(-\frac{1}{T_2} + \frac{1}{T_1}) $$

#### Кінцева формула

Об'єднуючи всі частини, отримуємо:

$$ Q = n[a_{(v)}(T_2 - T_1) + b_{(v)}\frac{T_2^2 - T_1^2}{2} + c_{(v)}(-\frac{1}{T_2} + \frac{1}{T_1})] $$

In [12]:
# 5. Аналітичний розрахунок теплоти
# Обчислюємо кількість теплоти за формулою Q = n·∫(T1 до T2) cv dT
# Інтеграл розбиваємо на три частини відповідно до членів формули cv = a_v + b_v·T + c_v·T⁻²

# Аналітичне інтегрування дозволяє отримати точну формулу для обчислення теплоти
# Для кожного члена формули теплоємності застосовуємо відповідну формулу інтегрування:
# ∫a·dT = a·(T2-T1)
# ∫b·T·dT = b·(T2²-T1²)/2
# ∫c·T⁻²·dT = c·(-1/T2+1/T1)

Q_analytical = n * (
        a_v * (T2 - T1) +                  # Інтеграл від константи a_v
        b_v * (T2**2 - T1**2) / 2 +        # Інтеграл від лінійного члена b_v·T
        c_v * (-1/T2 + 1/T1)               # Інтеграл від оберненого квадрата c_v·T⁻²
)

#### Детальний розгляд кожного доданка:

1. **a_v * (T2 - T1)** - інтегрування константи a_v
   - ∫a_v·dT = a_v·T|_(T1)^(T2) = a_v·(T2-T1)
   - Фізичний зміст: внесок постійної складової теплоємності в загальну теплоту
   - Цей член відображає базову кількість теплоти, необхідну для нагрівання газу, якщо б його теплоємність не залежала від температури

2. **b_v * (T2**2 - T1**2) / 2** - інтегрування лінійного члена b_v·T
   - ∫b_v·T·dT = b_v·T²/2|_(T1)^(T2) = b_v·(T2²-T1²)/2
   - Фізичний зміст: внесок лінійної залежності теплоємності від температури
   - Цей член враховує збільшення теплоємності з ростом температури через активацію коливальних ступенів свободи молекул

3. **c_v * (-1/T2 + 1/T1)** - інтегрування оберненого квадрата c_v·T⁻²
   - ∫c_v·T⁻²·dT = -c_v·T⁻¹|_(T1)^(T2) = c_v·(-1/T2+1/T1)
   - Фізичний зміст: внесок квантових ефектів та міжмолекулярної взаємодії
   - Цей член особливо важливий при низьких температурах, коли квантові ефекти суттєво впливають на теплоємність

#### Перевірка розмірностей для кожного доданка

Ця перевірка є важливим етапом розрахунку, який дозволяє виявити можливі помилки в формулах або одиницях вимірювання.

Для першого доданка a_v * (T2 - T1):
- a_v має розмірність [Дж/(моль·К)]
- (T2 - T1) має розмірність [К]
- Добуток має розмірність [Дж/моль]
- Перевірка: [Дж/(моль·К)] · [К] = [Дж/моль] ✓

Для другого доданка b_v * (T2**2 - T1**2) / 2:
- b_v має розмірність [Дж/(моль·К²)]
- (T2^2 - T1^2) має розмірність [К²]
- Добуток має розмірність [Дж/моль]
- Перевірка: [Дж/(моль·К²)] · [К²] = [Дж/моль] ✓

Для третього доданка c_v * (-1/T2 + 1/T1):
- c_v має розмірність [Дж·К²/(моль)]
- (-1/T2 + 1/T1) має розмірність [1/К²]
- Добуток має розмірність [Дж/моль]
- Перевірка: [Дж·К²/(моль)] · [1/К²] = [Дж/моль] ✓

Загальний результат Q_analytical має розмірність [Дж/моль] · [моль] = [Дж], що відповідає розмірності теплоти.

In [13]:
# Обчислення числового значення теплоти та переведення в кілоджоулі
# Ділення на 1000 переводить результат з джоулів у кілоджоулі
# Формат {Q_analytical/1000:.2f} означає виведення з двома знаками після коми
print(f"   Аналітичний метод: {Q_analytical/1000:.2f} кДж")

   Аналітичний метод: 162243.12 кДж


### Числовий розрахунок теплоти методом трапецій

#### Теоретична основа

При розрахунку кількості теплоти необхідно обчислити визначений інтеграл:

$$ Q = n\int_{T_1}^{T_2} c_v \, dT $$

де теплоємність залежить від температури:

$$ c_v = a_{(v)} + b_{(v)}T + c_{(v)}T^{-2} $$

Цей метод дозволяє обчислити інтеграл чисельно, без отримання аналітичної формули. Він особливо корисний для функцій, які складно або неможливо проінтегрувати аналітично.

#### Метод трапецій

Метод трапецій - це один з методів числового інтегрування, який апроксимує площу під кривою за допомогою трапецій. Основні кроки методу:

1. Розбиття інтервалу інтегрування [T₁, T₂] на N малих відрізків
2. На кожному відрізку апроксимація підінтегральної функції лінійною функцією
3. Обчислення площі трапеції для кожного відрізка
4. Сумування всіх площ, отримуючи наближене значення інтегралу

#### Формула методу трапецій

$$ \int_{T_1}^{T_2} f(T) \, dT \approx \frac{h}{2}[f(T_1) + 2\sum_{i=1}^{n-1}f(T_i) + f(T_n)] $$

або в розгорнутому вигляді:

$$ \int_{T_1}^{T_2} f(T) \, dT \approx \frac{h}{2}[f(T_1) + 2f(T_2) + 2f(T_3) + ... + 2f(T_{n-1}) + f(T_n)] $$

де:
- h - крок інтегрування: $h = \frac{T_2 - T_1}{n}$
- n - кількість відрізків
- f(T) - підінтегральна функція

#### Переваги методу трапецій:
- Простота реалізації
- Універсальність (працює з будь-якими функціями)
- Достатня точність для більшості практичних задач
- Можливість контролювати точність через вибір кількості кроків

In [14]:
# 6. Числовий розрахунок теплоти методом трапецій
# Створення масиву температур для числового інтегрування
steps = 1000  # кількість кроків інтегрування (більше кроків = вища точність)

# Функція np.linspace створює масив з рівномірно розподіленими точками
# Параметри: початкове значення, кінцеве значення, кількість точок
# Результат: масив [T₁, T₁+h, T₁+2h, ..., T₂-h, T₂], де h = (T₂-T₁)/(steps-1)
T = np.linspace(T1, T2, steps)  # рівномірне розбиття інтервалу [T1, T2]

# Визначення функції теплоємності як анонімної функції (lambda-функції)
# lambda T: ... - створює функцію, яка приймає аргумент T і повертає вираз після двокрапки
# Ця функція обчислює теплоємність cv для заданої температури T
# Використовуємо ту саму формулу cv = a_v + b_v·T + c_v/T², що і в аналітичному методі
cv = lambda T: a_v + b_v*T + c_v/(T*T)  # теплоємність як функція від T

# Числове інтегрування методом трапецій
# 1. [cv(t) for t in T] - створює список значень теплоємності для кожної температури
#    (це називається list comprehension - компактний спосіб створення списків)
# 2. np.trapezoid(y, x) - функція з бібліотеки NumPy, яка реалізує метод трапецій
#    y - масив значень функції (теплоємності)
#    x - масив значень аргументу (температури)
# 3. n * ... - множення результату інтегрування на кількість молів
#    (перехід від питомої теплоти до повної теплоти)
Q_numerical = n * np.trapezoid([cv(t) for t in T], T)

In [15]:
# Виведення результату числового розрахунку
print(f"   Числовий метод: {Q_numerical/1000:.2f} кДж")

   Числовий метод: 162243.09 кДж


#### Детальне пояснення реалізації методу трапецій

1. **Вибір кількості кроків (`steps = 1000`):**
   - Визначає точність обчислення: більше кроків = вища точність
   - Впливає на обчислювальну складність: більше кроків = більше обчислень
   - 1000 кроків забезпечує похибку порядку 0.01%, що достатньо для практичних задач
   - Для складніших функцій може знадобитися більша кількість кроків

2. **Створення масиву температур (`np.linspace(T1, T2, steps)`):**
   - `np.linspace` - функція для створення рівномірно розподілених точок
   - Аргументи: початкове значення, кінцеве значення, кількість точок
   - Результат: масив [T₁, T₁+h, T₁+2h, ..., T₂], де h = (T₂-T₁)/(steps-1)
   - Точки рівномірно розподілені між T₁ та T₂
   - Рівномірний розподіл точок оптимальний для методу трапецій
   - Для функцій з різкими змінами можна використовувати нерівномірний розподіл

3. **Визначення функції теплоємності (`cv = lambda T: a_v + b_v*T + c_v/(T*T)`):**
   - `lambda T: ...` - створює анонімну функцію з аргументом T
   - Компактний запис для простих функцій без необхідності окремого визначення
   - Обчислює теплоємність для заданої температури
   - Формула cv = a_v + b_v*T + c_v/(T*T) відповідає теоретичній залежності
   - Використовує коефіцієнти a_v, b_v, c_v з таблиці 1.1
   - Можна легко модифікувати для інших газів, змінивши коефіцієнти

4. **Обчислення значень функції (`[cv(t) for t in T]`):**
   - List comprehension - компактний спосіб створення списку
   - Обчислює значення cv для кожної температури з масиву T
   - Використовує list comprehension для компактності
   - Альтернативний запис через цикл:
     ```python
     values = []
     for t in T:
         values.append(cv(t))
     ```
   - List comprehension ефективніший і читабельніший

5. **Інтегрування методом трапецій (`np.trapezoid(y, x)`):**
   - `np.trapezoid` - функція з бібліотеки NumPy для інтегрування
   - Реалізує формулу методу трапецій: ∫f(x)dx ≈ (h/2)[f(x₀) + 2f(x₁) + ... + 2f(xₙ₋₁) + f(xₙ)]
   - Аргументи: масив значень функції (y), масив значень аргументу (x)
   - Автоматично обчислює крок інтегрування h на основі масиву x
   - Автоматично застосовує формулу методу трапецій
   - Враховує можливу нерівномірність розподілу точок

6. **Множення на кількість молів (`n * ...`):**
   - Множення результату інтегрування на кількість молів
   - Перехід від питомої теплоти (Дж/моль) до повної теплоти (Дж)
   - Відповідає фізичному змісту задачі: загальна теплота = питома теплота × кількість речовини

#### Оцінка точності методу трапецій

Теоретична похибка методу трапецій для функції f(x) з неперервною другою похідною f''(x) на інтервалі [a,b] оцінюється як:

$$ |Похибка| \leq \frac{(b-a)^3 \cdot M}{12n^2} $$

де M = max|f''(x)| на [a,b], а n - кількість відрізків.

У нашому випадку:
- Інтервал інтегрування: (T₂-T₁) = 975 К
- Кількість кроків: n = 1000
- Друга похідна функції cv(T): f''(T) = 2·c_v·(6/T⁴)

Оцінка максимального значення |f''(T)| на [T₁,T₂]:
- При T = T₁ = 298.15 К: |f''(T₁)| ≈ 12·|c_v|/T₁⁴ ≈ 1.5·10⁻⁹
- Похибка ≤ 975³·1.5·10⁻⁹/(12·1000²) ≈ 1.2·10⁻⁵ Дж/моль

Відносна похибка:
ε ≤ 1.2·10⁻⁵/(Q_analytical/n) ≈ 2.8·10⁻⁷ ≈ 0.000028%

Така мала похибка пояснює ідеальне співпадіння результатів аналітичного та числового методів з точністю до 2 знаків після коми.

Загалом, похибка методу трапецій пропорційна квадрату кроку інтегрування:

$$ \text{Похибка} \sim O(h^2) = O(\frac{(T_2-T_1)^2}{n^2}) $$

При n = 1000 кроків похибка становить менше 0.01%, що цілком достатньо для практичних розрахунків.

#### Переваги методу трапецій

1. Простота реалізації - метод легко програмується
2. Висока точність при достатній кількості кроків
3. Універсальність - працює з будь-якими функціями
4. Автоматичне врахування всіх особливостей функції
5. Можливість контролювати точність через вибір кількості кроків

#### Порівняння з аналітичним методом

Переваги числового методу:
- Універсальність (працює з будь-якими функціями, навіть тими, що не інтегруються аналітично)
- Простота реалізації (не потрібно виводити формули інтегралів)
- Легко модифікувати для інших функцій (достатньо змінити визначення функції cv)
- Можливість візуалізації процесу інтегрування

Недоліки числового методу:
- Наявність похибки округлення (хоча при достатній кількості кроків вона мінімальна)
- Потребує більше обчислювальних ресурсів (особливо при великій кількості кроків)
- Результат залежить від кількості кроків (потрібно контролювати збіжність)
- Не дає аналітичної формули для подальшого аналізу

In [16]:
# 7. Порівняння результатів аналітичного та числового методів
# Цей блок коду порівнює результати обох методів та обчислює різницю між ними

# Виведення заголовка розділу
print(f"\n3. Результати розрахунку теплоти:")

# Виведення результату аналітичного методу в кілоджоулях
# Ділення на 1000 переводить джоулі в кілоджоулі
print(f"\tАналітичний метод: {Q_analytical/1000:.2f} кДж")

# Виведення результату числового методу в кілоджоулях
print(f"\tЧисловий метод: {Q_numerical/1000:.2f} кДж")

# Обчислення та виведення абсолютної різниці між результатами методів
# abs() - функція для обчислення абсолютного значення (модуля) числа
# Формат {:.4f} виводить число з 4 знаками після коми
print(f"\tРізниця між методами: {abs(Q_analytical - Q_numerical)/1000:.4f} кДж")

# Обчислення та виведення відносної похибки (у відсотках)
# Відносна похибка = |Різниця|/|Точне значення|·100%
# Аналітичний метод приймаємо за точне значення
relative_error = abs(Q_analytical - Q_numerical)/abs(Q_analytical)*100
print(f"\tВідносна похибка: {relative_error:.6f}%")


3. Результати розрахунку теплоти:
	Аналітичний метод: 162243.12 кДж
	Числовий метод: 162243.09 кДж
	Різниця між методами: 0.0206 кДж
	Відносна похибка: 0.000013%


#### Інтерпретація результатів:
- Майже ідеальне співпадіння результатів (різниця менше 0.1 кДж)
- Підтверджує правильність обох методів розрахунку
- Демонструє високу точність методу трапецій при достатній кількості кроків
- Для даної задачі обидва методи дають практично ідентичні результати

#### Практичні висновки:
1. Для простих функцій, які легко інтегруються аналітично, аналітичний метод є кращим
    (дає точний результат з мінімальними обчисленнями)
2. Для складних функцій числовий метод є єдиним практичним рішенням
3. При розробці програмного забезпечення числовий метод часто є кращим вибором
    через його універсальність та простоту реалізації

# Висновки

## Практична цінність розрахунку

Розрахунок теплоти, що поглинає діоксид вуглецю при нагріванні, має значну практичну цінність для металургійної промисловості, зокрема для конвертерного виробництва сталі:

1. **Оптимізація теплових балансів** — точне визначення кількості теплоти дозволяє ефективніше розподіляти енергетичні ресурси в конвертерних агрегатах, що призводить до зниження енергоспоживання.

2. **Проєктування обладнання** — розуміння теплових властивостей CO₂ необхідне для правильного розрахунку теплообмінників, систем охолодження та інших елементів конвертерних агрегатів.

3. **Контроль технологічних процесів** — знання кількості теплоти, що поглинається або виділяється, дозволяє точніше контролювати температурні режими плавки, що впливає на якість кінцевого продукту.

4. **Екологічні аспекти** — розрахунки теплових ефектів допомагають оптимізувати процеси з точки зору зменшення викидів CO₂ та інших парникових газів.

5. **Економічна ефективність** — зниження енерговитрат завдяки точним розрахункам безпосередньо впливає на собівартість продукції та конкурентоспроможність підприємства.

## Порівняння методів розв'язання

У роботі було використано два методи розрахунку теплоти:

### 1. Аналітичний метод

**Переваги:**
- Дає точний результат без похибок округлення
- Потребує менше обчислювальних ресурсів
- Дозволяє отримати аналітичну формулу для подальшого аналізу
- Забезпечує глибше розуміння фізичних процесів

**Недоліки:**
- Вимагає виведення формул інтегралів, що може бути складним для функцій з нетривіальною залежністю від температури
- Не універсальний — працює лише з функціями, які можна проінтегрувати аналітично
- Складніший у реалізації для нестандартних залежностей теплоємності від температури

### 2. Числовий метод (метод трапецій)

**Переваги:**
- Універсальність — працює з будь-якими функціями, навіть тими, що не інтегруються аналітично
- Простота реалізації — не потрібно виводити формули інтегралів
- Легко модифікується для інших функцій
- Можливість візуалізації процесу інтегрування

**Недоліки:**
- Наявність похибки округлення (хоча при достатній кількості кроків вона мінімальна)
- Потребує більше обчислювальних ресурсів, особливо при великій кількості кроків
- Результат залежить від кількості кроків (потрібно контролювати збіжність)

### Порівняння результатів

Для розглянутої задачі обидва методи дали практично ідентичні результати:
- Аналітичний метод: 162243.12 кДж
- Числовий метод: 162243.09 кДж
- Різниця між методами: 0.0206 кДж
- Відносна похибка: 0.000013%

Така висока точність числового методу пояснюється:
1. Достатньою кількістю кроків інтегрування (1000)
2. Гладкістю функції теплоємності CO₂ в заданому діапазоні температур
3. Відносно простою формою залежності теплоємності від температури

## Загальні висновки

1. **Методологічний аспект** — розглянута задача демонструє ефективність як аналітичного, так і числового підходів до розрахунку теплових ефектів. Для інженерних розрахунків у металургії доцільно використовувати обидва методи: аналітичний — для стандартних задач з відомими формулами, числовий — для складних залежностей теплоємності від температури.

2. **Фізичний зміст результату** — отримане значення теплоти (162243.12 кДж) показує, що для нагрівання 100 м³ діоксиду вуглецю від 25°C до 1000°C при сталому об'ємі потрібна значна кількість енергії, еквівалентна спалюванню приблизно 3 кг природного газу. Це підкреслює важливість ефективного використання теплової енергії в металургійних процесах.

3. **Практичне застосування** — розроблений алгоритм розрахунку може бути інтегрований у системи автоматизованого керування конвертерними агрегатами для оптимізації теплових режимів плавки, що сприятиме підвищенню якості продукції та зниженню енерговитрат.

4. **Освітній аспект** — поєднання теоретичного матеріалу з практичним розрахунком та програмною реалізацією створює цілісну методологічну базу для вивчення термодинамічних процесів у металургії.

5. **Перспективи розвитку** — подальші дослідження можуть бути спрямовані на врахування реальних умов конвертерного процесу, зокрема нерівномірності нагрівання, впливу домішок у газовій фазі та взаємодії з іншими компонентами системи.

Таким чином, проведений розрахунок не лише демонструє методику визначення теплових ефектів, але й має безпосереднє практичне значення для оптимізації роботи конвертерних агрегатів, що є важливим кроком у підвищенні ефективності металургійного виробництва та зменшенні його екологічного впливу.